app/controllers/products_controller.rb

```diff
@@ -1,29 +1,57 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only:[:show, :edit, :update, :destroy]
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
   
   def index
+    @products = Product.all
   end
-  
-  def show
-  end
-  
+
+  def show; end
+
   def dashboard
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
   end
 
   def create
+    product = Product.create!(product_params.except(:image))
+    product.image.attach(product_params[:image])
+      
+    redirect_to dashboard_path
+    
+    rescue ActiveRecord::RecordInvalid
+      redirect_to new_error_product_path
+
```

Comment:
remove the empty line here. Also indentation looks a bit weird. `rescue` should be aligned with `def`